kubemarine/procedures/check_paas.py

```diff
@@ -309,9 +313,45 @@ def thirdparties_hashes(cluster):
                 continue
 
             if 'sha1' not in config:
-                # silently skip if SHA not defined
-                continue
-
+                is_curl = config['source'][:4] == 'http' and '://' in config['source'][4:8]
+                expected_sha = None
+
+                # Get sha from source, if it can be downloaded
+                if is_curl:
+                    cluster.log.verbose(f"Thirdparty {path} doesn't have default sha, download it...")
+                    #Create tmp dir for loading thirdparty without default sha
+                    random_dir = "/tmp/%s" % uuid.uuid4().hex
+                    final_commands = "rm -r -f %s" % random_dir
+                    random_path = "%s%s" % (random_dir, path)
+                    cluster.log.verbose('Temporary path: %s' % random_path)
+                    remote_commands = "mkdir -p %s" % ('/'.join(random_path.split('/')[:-1]))
+                    #Load thirdparty to temporary dir
+                    remote_commands += "&& sudo curl -f -g -s --show-error -L %s -o %s" % (config['source'], random_path)
+                    results = first_control_plane.sudo(remote_commands, hide=True, warn=True)
+                    for host, result in results.items():
+                        if result.failed:
+                            broken.append(f"Can`t download thirdparty {path} on {host.host} for getting sha: {result.stderr}")
+                    if not results.is_any_failed():
+                        #Get temporary thirdparty sha
+                        cluster.log.verbose(f"Get temporary thirdparty sha...")
+                        results = first_control_plane.sudo(f'openssl sha1 {random_path} | sed "s/^.* //"', warn=True)
+                        for host, result in results.items():
+                            if result.failed:
+                                broken.append(f'failed to get sha for temporary file {random_path} on {host.host}: {result.stderr}')
+                            else:
+                                expected_sha = result.stdout.strip()
+                                cluster.log.verbose("Expected sha was got")
+                    # Remove temporary dir in any case
+                    cluster.log.verbose(f"Remove temporary dir {random_dir}...")
+                    first_control_plane.sudo(final_commands, hide=True, warn=True)
+
+                if expected_sha is None:
+                    cluster.log.verbose(f"Can`t get expected sha for {path}, skip it")
+                    #Skip checking sha if something went wrong or this sha can't be loaded
+                    continue
```

Comment:
For default thirdparties, we might additionally compare computed expected_sha with the recommended default sha and raise warning if they are not equal.

Reply:
Sorry, probably I said not accurate enough. I meant sha of default thirdparties which we recommend for each kubernetes version. For example [kubeadm](https://github.com/Netcracker/KubeMarine/blob/0.7.1/kubemarine/resources/configurations/defaults.yaml#L82). We can compare that sha with calculated expected_sha on first_control_plane.

In other words, let's assume user specified custom source for `kubeadm` and not specified custom sha. In this case config['sha'] will be removed. We expect that it is different source but the same thirdparty version, because this version is QA tested. Other versions are allowed but we can warn if they are used. @koryaga do we need to do that?

Reply:
@ilia1243 recommended 3dparty version was calculated additionally in paas_check. Please check